File: ios/Models/HUDSession.swift
import Foundation

struct HUDSession: Codable, Identifiable {
    let id: UUID
    let platform: String
    let deviceModel: String
    let osVersion: String
    let startTimestamp: Date
    var endTimestamp: Date?
    var thermalEvents: [ThermalEvent]
    var averageFPS: Double
    var latencyMsP50: Double
    var latencyMsP90: Double
    var batteryDeltaPercent: Double
    var fallbackTriggered: Bool
    var offlineDurationMs: Int
}

File: android/app/src/main/java/com/golfiq/hud/model/HUDSession.kt
package com.golfiq.hud.model

import java.time.Instant
import java.util.UUID

data class HUDSession(
    val id: UUID,
    val platform: String,
    val deviceModel: String,
    val osVersion: String,
    val startTimestamp: Instant,
    val endTimestamp: Instant?,
    val thermalEvents: List<ThermalEvent>,
    val averageFps: Double,
    val latencyMsP50: Double,
    val latencyMsP90: Double,
    val batteryDeltaPercent: Double,
    val fallbackTriggered: Boolean,
    val offlineDurationMs: Int,
)